src/main/sensors/battery.c

```diff
@@ -187,10 +190,10 @@ void batteryUpdatePresence(void)
         batteryCriticalVoltage = batteryCellCount * batteryConfig()->vbatmincellvoltage;
     } else if (
         voltageState != BATTERY_NOT_PRESENT
-        && voltageMeter.filtered <= batteryConfig()->batteryNotPresentLevel
         && isVoltageStable
+        && !isVoltageFromBat
```

Comment:
Have you tested what happens when the battery is disconnected, and the voltage on the filter cap drops from the '2S+ range' into the 'USB range' and then into the '1S range'?